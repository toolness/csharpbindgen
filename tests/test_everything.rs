use std::fs;
use std::path::PathBuf;
use insta::assert_snapshot_matches;

use csharpbindgen::{
    Builder,
    CSAccess
};

fn run_example_test<F: FnOnce(Builder) -> Builder>(name: &'static str, build: F) {
    let builder = build(Builder::new("MyDll", load_example(name)));
    assert_snapshot_matches!(name, builder.generate().unwrap());
}

fn load_example(name: &'static str) -> String {
    let mut path = PathBuf::new();
    path.push("tests");
    path.push("rust");
    path.push(format!("{}.rs", name));
    fs::read_to_string(path).unwrap()
}

#[test]
fn test_main_example() {
    run_example_test("main_example", |b|
        b.class_name("MyStuff")
         .ignore(&["ignore_*", "IGNORE_*", "Ignore*"])
         .access("public_func", CSAccess::Public)
         .access("PublicStruct", CSAccess::Public)
    );
}

#[test]
fn test_safe_handles() {
    run_example_test("safe_handles_example", |b| b.use_safe_handles());
}
