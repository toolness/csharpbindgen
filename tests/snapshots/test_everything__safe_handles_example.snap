---
created: "2019-07-14T11:57:45.370831700Z"
creator: insta@0.8.1
source: tests/test_everything.rs
expression: builder.generate().unwrap()
---
// This file has been auto-generated, please do not edit it.

using System;
using System.Runtime.InteropServices;

internal class RustExports {
    [DllImport("MyDll")]
    internal static extern byte blarg(MyOpaqueStructHandle c);

}

